### DISCUSSION TITLE
Support for gtk themes

### DISCUSSION THREAD

**Discussion by mr-cheffy** [zen-browser/desktop#760] [Ideas]
### What feature would you like to see?

It may be nice to bring Gtk support for Gnome and similar desktop systems, so that your project will be more compatible as a theme on linux devices.

![Screenshot from 2024-08-25 23-41-39](https://github.com/user-attachments/assets/d37b3e9e-a4be-4ef0-81dc-9d533a323628)

 
 ~ _Created automatically from issue: https://github.com/zen-browser/desktop/issues/697_

**Comment by SomeDudeBrosky** [comment]
This would Put unnecessary strain on an unpaid developer and add no value to the end product.

**Reply by rznkolds** [reply]
> This would Put unnecessary strain on an unpaid developer and add no value to the end product.

I do not agree with you on this issue, many products offer themselves by default for linux applications, what is missing from zen browser that it deprives itself of this situation? Apart from this situation, it makes the product more attractive, personally, every linux user wants a browser compatible with their own theme, so if they proceed with the logic that the program just works, it will be a program for me to just look at and exit, I'm not just talking about the theme here, I'm talking about the compatibility of the product with the system, apart from that, if a few more topics are opened, I will agree again.

You can think of this situation as if all the wooden pencils in a pencil case stand straight and one stands upside down, in addition to this, the product can help in the open source community, which can help in this regard, and it does not create much burden to make gtk themes compatible with a product.

I think it was a sensible thing to bring up.

**Reply by dnut** [reply]
If theming adds no value, then why did zen implement custom light and dark themes? What a useless comment.

**Reply by rznkolds** [reply]
You realize you're talking nonsense right now, right ?

**Reply by dnut** [reply]
rznkolds, I was responding to the person who started this thread.

**Reply by rznkolds** [reply]
> rznkolds, I was responding to the person who started this thread.

Oh sorry then.

**Comment by dnut** [comment]
This is a no-brainer. Why bother with custom themes made just for zen, when you can tap into the entire gtk ecosystem with one fell swoop? The user already chose a theme that they like for their system. Why violate the user's intention?

The current theming options are terrible and are one of the primary reasons I'm keeping firefox+sidebery as my primary browser. Dark is too dark, light is too light, and custom themes don't work. Just support the native OS theming, like any well-behaved application.

**Comment by D-Brox** [comment]
Open `about:config` and set `browser.tabs.inTitlebar` to 0 so it uses the GTK window decoration

**Reply by dnut** [reply]
This does not apply the gtk theme to the browser. All this does is add an extra titlebar to the top. Yes the titlebar is gtk themed. But the rest of the browser retains the zen theme.

**Comment by kRHYME7** [comment]
Firefox follows gtk using `System theme - auto`  and  Google Chrome now has `GTK/QT` option. 

Can we discuss this further? Will it break zen-browser if we support GTK theming like firefox (and thunderbird) does?

**Comment by hexqubit** [comment]
Look the dislikes 🙂. Please update

**Comment by Andndre** [comment]
<img width="1308" height="333" alt="image" src="https://github.com/user-attachments/assets/562171f0-7fcc-4861-9b74-84468eda3093" />

Using [Catppuccin-GTK-Theme](https://github.com/Fausto-Korpsvart/Catppuccin-GTK-Theme) theme, this is how it looks now.

**Comment by nms808** [comment]
Native Firefox theming was stripped from Zen. Good luck on that front, but I too shall cast my vote for GTK themes for Linux. Lucky for me I have enough knowledge (and time!) to go in and make changes to code bases for my own personal customizations on any OS out there that can run Zen. I suggest you attempt learning to do the same, at the very least for your OS of choice if you want true satisfaction. More often then not, it's the only way to fly around these parts, lest you be GWTW! 

![GWTW](https://github.com/user-attachments/assets/add29a2f-63b9-481a-b7fa-4096908d2893)
